graphics/fpga-graphics/verilator/main_beam.cpp

```diff
@@ -87,12 +86,25 @@ int main(int argc, char* argv[]) {
 
         // update texture once per frame at start of blanking
         if (top->sy == V_RES && top->sx == 0) {
+            // check for quit event
+            SDL_Event e;
+            if (SDL_PollEvent(&e)) {
+                if (e.type == SDL_QUIT) {
+                    break;
+                }
+            }
+
             SDL_UpdateTexture(sdl_texture, NULL, screenbuffer, H_RES*sizeof(Pixel));
```

Comment:
According to the docs https://wiki.libsdl.org/SDL_UpdateTexture:

> This is a fairly slow function, intended for use with static textures that do not change often.
>
> If the texture is intended to be updated often, it is preferred to create the texture as streaming and use the locking functions referenced below. While this function will work with streaming textures, for optimization reasons you may not get the pixels back if you lock the texture afterward.

I tried getting this to work with `SDL_TEXTUREACCESS_STREAMING`, `SDL_LockTexture`, and `SDL_UnlockTexture`, but I couldn't get it to work. Might be worth looking into further later, though, for another perf gain. But the current code hits 60 FPS, so I didn't want to spend any more time on it.